functions/transformations/group_test.go

```diff
@@ -590,3 +592,133 @@ func TestGroup_Process(t *testing.T) {
 		})
 	}
 }
+
+func TestMergeGroupRule(t *testing.T) {
+	var (
+		from      = &inputs.FromProcedureSpec{}
+		groupNone = &transformations.GroupProcedureSpec{
+			GroupMode: functions.GroupModeBy,
+			GroupKeys: []string{},
+		}
+		groupBy = &transformations.GroupProcedureSpec{
+			GroupMode: functions.GroupModeBy,
+			GroupKeys: []string{"foo", "bar", "buz"},
+		}
+		groupExcept = &transformations.GroupProcedureSpec{
+			GroupMode: functions.GroupModeExcept,
+			GroupKeys: []string{"foo", "bar", "buz"},
+		}
+		filter = &transformations.FilterProcedureSpec{}
+	)
+
+	tests := []plantest.RuleTestCase{
+		{
+			Name:  "single group",
+			Rules: []plan.Rule{&transformations.MergeGroupRule{}},
+			Before: &plantest.PlanSpec{
+				Nodes: []plan.PlanNode{
+					plan.CreateLogicalNode("from", from),
+					plan.CreateLogicalNode("group", groupBy),
+				},
+				Edges: [][2]int{
+					{0, 1},
+				},
+			},
+			After: &plantest.PlanSpec{
```

Comment:
I just added a concise way to express that a rule should not apply to a plan, by omitting the `After` field and specifying `NoChange: true`.